modules/persistent-storage-csi-vsphere-top-aware.adoc

```diff
@@ -0,0 +1,171 @@
+// Module included in the following assemblies:
+//
+// persistent-storage-csi-vsphere.adoc
+//
+
+:content-type: PROCEDURE
+[id="persistent-storage-csi-vsphere-top-aware_{context}"]
+= vSphere Topology Awareness
+
+{product-title} provides the ability to deploy {product-title} for vSphere on different zones and regions, which allows you to deploy over multiple data centers and clusters, thus helping to avoid a single-point-of-failure.
+
+:FeatureName: vSphere topology Awareness
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+.Procedure
+Use the following procedure to deploy {product-title} for vSphere storage on different zones and regions.
+
+. In the VMware vCenter vSphere client GUI:
+.. Create and assign `zone` and `region` type tags as needed to the hosts on the desired clusters.
+.. Create and apply `cat-zone-local-datastore` tags to the desired datastores.
+
++
+For more information about vSphere tags, see the WMware vSphere documentation.
+
+. In the `clustercsidriver` object YAML file, edit the `driverConfig` section:
+* Specify the `zone` and `region` tags that you created earlier.
+* Set `driverType` to `vSphere`.
++
+[source, terminal]
+----
+~ $ oc get clustercsidriver csi.vsphere.vmware.com -o yaml
+----
++
+Example output:
++
+[source, terminal]
+----
+apiVersion: operator.openshift.io/v1
+kind: ClusterCSIDriver
+metadata:
+ name: csi.vsphere.vmware.com
+spec:
+ logLevel: Normal
+ managementState: Managed
+ observedConfig: null
+ operatorLogLevel: Normal
+ unsupportedConfigOverrides: null
+  driverConfig: 
+    driverType: vSphere <1>
+    vSphere:
+      topologyCategories: <2>
+        - k8s-zone
+        - k8s-region
+----
+<1> Ensure that `driverType` is set to `vSphere`.
+<2> `Zone` and `region` tags created earlier in vCenter.
+
+. Verify that `CSINode` object has topology keys:
++
+[source, terminal]
+----
+~ $ oc get csinode
+----
++
+Example output:
++
+[source, terminal]
+----
+NAME                        DRIVERS     AGE
+co8-4s88d-infra-2m5vd       1           27m
+co8-4s88d-master-0          1           70m
+co8-4s88d-master-1          1           70m
+co8-4s88d-master-2          1           70m
+co8-4s88d-worker-j2hmg      1           47m
+co8-4s88d-worker-mbb46      1           47m
+co8-4s88d-worker-zlk7d      1           47m
+----
++
+[source, terminal]
+----
+~ $ oc get csinode co8-4s88d-worker-j2hmg -o yaml
+----
++
+Exmaple output:
++
+[source, terminal]
+----
+...
+spec:
+    drivers:
+    - allocatable:
+        count: 59
+    name: csi-vsphere.vmware.com
+    nodeID: co8-4s88d-worker-j2hmg
+    topologyKeys: <1>
+    - topology.csi-vmware.com/k8s-zone
+    - topology.csi-vmware.com/k8s-region
+----
```

Comment:
We will have to update this to reflect `openshift-zone` and `openshift-region`

Reply:
Done